src/sidebar/components/NotebookView.js

```diff
@@ -36,22 +57,51 @@ function NotebookView({ loadAnnotationsService }) {
     }
   }, [loadAnnotationsService, focusedGroup, store]);
 
-  const rootThread = useRootThread();
-  const groupName = focusedGroup?.name ?? '…';
+  // Pagination-page-changing callback
+  const onChangePage = newPage => {
+    setPaginationPage(newPage);
+  };
+
+  // When filter values or focused group are changed, reset pagination to page 1
+  useEffect(() => {
+    onChangePage(1);
+  }, [filters, focusedGroup]);
+
+  // Scroll back to here when pagination page changes
+  const threadListScrollTop = useRef(/** @type {HTMLElement|null}*/ (null));
+  useLayoutEffect(() => {
+    // If pagination page has changed, scroll to top of list of page of threads
+    // to semi-represent a new page load
+    // TODO: Transition and effects here should be improved
+    if (paginationPage !== lastPaginationPage.current) {
+      scrollIntoView(threadListScrollTop.current);
+      lastPaginationPage.current = paginationPage;
+    }
+  }, [paginationPage]);
```

Comment:
Removing this check causes this test to fail:

https://github.com/hypothesis/client/pull/2994/files#diff-6ed49c32a22a0a11ea2cac90355a4af9e8a312adee7a926b72724afa9515521fR121

Which is odd—`setPaginationPage` is called twice with the same (primitive) value...I'm not sure why it's triggering a `useEffect` that should only trigger when `paginationPage` is _changed_.

Reply:
>  Which is odd—setPaginationPage is called twice with the same (primitive) value...I'm not sure why it's triggering a useEffect that should only trigger when paginationPage is changed.

Ah, I see the issue. It is to do with the first time the effect runs after the initial render, when `paginationPage` is 1. In that case when the check against `lastPaginationPage.current` is present the scroll is skipped because it is initialized to 1. This is what we want because there is no need to scroll to the top after the first render.

OK, so it turns out there is a need for this check even though the effect has only one dependency. Let's leave this as-is.